src/Engine/GameAction.java

```diff
@@ -956,7 +896,7 @@ public GameEventQueue getEvents(MapMaster gameMap)
       if( transformEvents.size() > 0 ) // if we successfully made a move action
       {
         GameEvent moveEvent = transformEvents.peek();
-        if (moveEvent.getEndPoint().equals(waitLoc)) // make sure we shouldn't be pre-empted
+        if (moveEvent.getEndPoint().equals(getMoveLocation())) // make sure we shouldn't be pre-empted
         {
           transformEvents.add(new UnitTransformEvent(actor, type.destinationType));
         }
```

Comment:
I feel like it might make sense to just add a `shouldPreempt()` function to `MoveAction` and base some or all of the other action types off of it to minimize validation duplication.

Reply:
Hmmm, maybe. Out of scope for this PR though.

Reply:
Made an issue for it.